journals/apps/search/backend.py

```diff
@@ -106,6 +97,8 @@ def add_item(self, item):
         # Get mapping
         mapping = self.mapping_class(item.__class__)
         if mapping.get_document_type() == JOURNAL_DOCUMENT_TYPE:
+            # sometime pipeline is missing. adding them to make sure they exists before using them in indexing)
+            self.add_ingest_pipeline()
```

Comment:
what cases is pipeline missing? Does not seem correct to add the pipeline every time we index a document. It used to be done in the put() method which gets called at index creation time which is when it should get set.

Reply:
In travis testing, when an item was adding to index[please see next line in same function], it was throwing error for missing pipeline. now every time when code tries to add new item into index, first it checks whether pipeline exists or not. if not exists it add one. I was not working there in put() method

Reply:
ok